doc/source/how-to-use-clientapp.rst

```diff
@@ -0,0 +1,303 @@
+
+To use :code:`CifarClient()` with the high-level :code:`ClientApp` API, we need to do several things:
+
+#. Wrap :code:`CifarClient` in a function, such as :code:`client_fn`. This allows multiple instances of :code:`CifarClient` to be created when :code:`client_fn` is called.
+#. Convert :code:`CifarClient` to a :code:`Client` implementation using the :code:`to_client()` method.
+
+.. code-block:: python
+
+    # file: client.py
+
+    def client_fn(cid: str) -> Client:
+        return CifarClient().to_client()
+
+Then, we create the :code:`ClientApp` as follows:
+
+.. code-block:: python
+
+    # file: client.py
+
+    app = ClientApp(client_fn=client_fn)
+
+Finally, your :code:`ClientApp` is ready to be executed!
+
+To run :code:`ClientApp` from CLI, use the |flower_clientapp_link|_ command. Pass the :code:`<module>:<attribute>` to the command, where :code:`module` is the filename (:code:`client.py`) and :code:`attribute` is the instantiated :code:`ClientApp` in the :code:`module`:
+
+.. code-block::shell
+
+    $ flower-client-app client:app  --insecure
+
+.. admonition:: Note
+    :class: note
+
+    In this example, the :code:`--insecure` command line argument starts Flower without HTTPS and is only used for prototyping. To run with HTTPS, we instead use the argument :code:`--certificates` and pass the paths to the certificates. Please refer to `Flower CLI reference <ref-api-cli.html>`_ for implementation details.
+
+As you can see, we can easily reuse existing Flower clients with the :code:`ClientApp` by adding 3 lines of code! Let’s now walk through how to use the low-level API for greater implementation flexibility in our projects.
+
+Using low-level API
+-------------------
+
+With Flower 1.8, we provide a set of low-level APIs to allow more versatile ways to implement any functionality that we like in the :code:`ClientApp`. The three functions that can be registered in a :code:`ClientApp` are:
+
+* :code:`app.train()`
+* :code:`app.evaluate()`
+
+which runs the training and evaluation on client-side data. 
+
+.. admonition:: Important
+    :class: important
+
+    As we continuously enhance Flower at a rapid pace, we'll periodically update the functionality and this how-to document. Please feel free to share any feedback with us!
+
+Next, we have a key part in the low-level API, which is the message passing format. In Flower 1.8, we introduced |message_link|_ and |context_link|_ objects. These are abstractions that unify the message formats that are relayed back and forth in a Flower project. When compared to the high-level example above, objects such as :code:`config` and :code:`parameter` need to be separately configured and tracked in each method (such as :code:`get_parameters()`, :code:`set_parameters()`, :code:`fit()`, and :code:`evaluate()`) which can make the workflow un-intuitive. To simplify the usage, we unify the data format via the :code:`Message` and :code:`Context` abstractions.
```

Comment:
A coupe of suggestions:
- "[messages, context] are abstractions to unify the message formats" --> i think it would be confusing to say the context is part of messaging format. This is because the context is local to the `SuperNode`, never leaves it. I think of it as some form of "in memory state" that persists until the`SuperNode` is shutdown. (in the case of simulations it persists until the simulation is over)
- "[...] which can make the workflow un-intuitive." -> i think the main limitation is that the workflow is a bit rigid. maybe we can adjust the wording ? "intuitive" is a bit relative 😅  (to me is very intuitive).

Reply:
@jafermarq 
* > This is because the context is local to the SuperNode

  Thanks for clarifying, I had the `LegacyContext` example in mind when I wrote this. I'll reword to clarify that context is the local state of the `SuperNode.
*   Sure, I'll reword too 😄